apps/prairielearn/src/pages/generatedFilesQuestion/generatedFilesQuestion.ts

```diff
@@ -48,6 +48,8 @@ export default function (options = { publicEndpoint: false }) {
         variant,
         res.locals.question,
         res.locals.course,
+        // res.locals.user.user_id is not always available during shared question preview.
+        res.locals.user ? res.locals.user.user_id : res.locals.authn_user,
```

Comment:
As noted in the comment, `user.user_id` was not always available here. Instead I am using the authn_user in these cases which should only occur during question sharing preview. In those cases, I don't think there should be a difference between `authn_user` and the current user as you would need to be authorized as an instructor to reach this.

Reply:
Interesting... under what conditions would we not have a `user` here?

Reply:
There is not a `user` attached to `res.locals` when trying to download files from `generatedFilesQuestion` in the `questionSharingPreview`. To reproduce, change this line to just `res.locals.user.user_id` and run the `questionSharingPreview` test.

Reply:
Sure, I can see that that's the case, my question was about _why_ we're missing `res.locals.user` there. I dug in a bit, turns out it's because `res.locals.user` is only assigned by the `authzCourseOrInstance` middleware, which doesn't run for that route.

I'd make the following changes (note that passing `authn_user` where an ID is expected isn't valid):

```suggestion
        // `res.locals.user` isn't populated for publicly-shared question previews.
        res.locals.user?.user_id ?? res.locals.authn_user.user_id,
```

Reply:
Got it. Committed the suggested code above. Thank you for the cleanup here!